lang/gcc7/Portfile

```diff
@@ -369,6 +386,17 @@ build.env-append \
 # we probably also want to clear out the rest of the configure.XYZ flags
 configure.optflags
 
+# gcc i386 misconfigures if our cctools as modification sends it to clang for assembly
+# export suitable ENVVARS to the build to force the traditional expected gas behaviour
+# have gcc build pass these ENVARS to the build process where it is needed
+if {${build_arch} eq "i386"} {
+    configure.env-append DISABLE_MACPORTS_AS_CLANG_SEARCH
+    build.env-append     DISABLE_MACPORTS_AS_CLANG_SEARCH
```

Comment:
probably just setting this in the configure.env is enough, as the build.env inherits from that. I just haven't tested it that way, so if you have any enthusiasm, please help test that.